### PR TITLE
Move Ldap Controls to another file

### PR DESCRIPTION
**Pull request by irvingoujAtDevolution**
organize the Ldap Control code to another file, makes it easier to add more control sequence in the future

### CODE REVIEW

File: proto/src/control.rs [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+            3 => {
+                let v = seq.pop();
+                let c = seq.pop();
+                let o = seq.pop();
+                (o, c, v)
+            }
+            _ => return Err(LdapProtoError::ControlSeqLen),
+        };
+
+        // trace!(?oid_tag, ?criticality_tag, ?value_tag);
```
Comment (CBenoit) [inline proto/src/control.rs:156]: Nitpick: either uncomment or delete this line

**Review by CBenoit — COMMENTED** [review]
Just moving code around, quite uncontroversial

Note: avoid commit messages such as "commit"

**Review by Firstyear — APPROVED** [review]

**Comment by Firstyear** [comment]
> Just moving code around, quite uncontroversial
> 
> Note: avoid commit messages such as "commit"

If there are other library structure changes you want to see, please lmk. I'm very open to them, as this library has "evolved" more than "was planned" so it shows that in it's structure.

File: proto/src/control.rs (hunk 1 above)
Comment (irvingoujAtDevolution) [reply proto/src/control.rs]: updated

**Review by Firstyear — APPROVED** [review]

**Comment by Firstyear** [comment]
@irvingoujAtDevolution No problem at all, you're doing great work here!

**Pull request merged**